### PR TITLE
RFC : Riemann-consul : Sends consul services status to riemann

### PR DESCRIPTION
**Pull request by shanielh**
Currently it's a basic tool thats changes the status of a service registered to consul if all the health checks inside it are not "ok".

I hope that people that use consul and riemann would have some suggestions to add.

### CODE REVIEW

File: bin/riemann-consul [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  def tick
+
+    leader = get(@consul_leader_url)
+    leader_hostname = URI.parse("http://" + leader[1..-2]).hostname
+
+    if (leader_hostname == @hostname)
+      services = JSON.parse(get(@consul_services_url))
+      for service in services
+        service_name = service[0]
```
Comment (aphyr) [inline bin/riemann-consul]: The only thing I'd like to ensure here is to make sure that all these services start with the same word, e.g. "consul whatever", "consul foobar", and so on. That way we don't clobber other services by accident. :)

Comment (shanielh) [inline bin/riemann-consul]: Oh, they don't. Do you want me to add "consul" as a prefix?

Comment (aphyr) [inline bin/riemann-consul]: That'd be great, thanks! Most folks use riemann to monitor multiple things so having prefixes keeps things easy to organize. :-)
On Oct 5, 2015 9:39 AM, Shani Elharrar <notifications@github.com> wrote:In bin/riemann-consul:
> +    report(opts)
> +  end
> +
> +  def get(url)
> +    Net::HTTP.get_response(url).body
> +  end
> +
> +  def tick
> +
> +    leader = get(@consul_leader_url)
> +    leader_hostname = URI.parse("http://" + leader[1..-2]).hostname
> +
> +    if (leader_hostname == @hostname)
> +      services = JSON.parse(get(@consul_services_url))
> +      for service in services
> +        service_name = service[0]

Oh, they don't. Do you want me to add "consul" as a prefix?

Comment (shanielh) [inline bin/riemann-consul]: Yeah. I do that too, I just use tags to keep it organized. I'll add a configurable prefix that will be "defaulted" to "consul."

Comment (aphyr) [inline bin/riemann-consul]: Hey, any chance you could make that change for me? Would love to get this merged. :)

Comment (shanielh) [inline bin/riemann-consul]: Did it (pushed w/ force to update in same commit).

Also added node status checking.

**Comment by aphyr** [comment]
Good to go! Thanks @shanielh! :)

**Pull request merged**